src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -9,19 +9,31 @@
 import pathlib
 
 from middlewared.event import EventSource
-from middlewared.schema import accepts, Bool, Dict, Float, Int, List, Ref, returns, Path, Str
-from middlewared.service import private, CallError, filterable_returns, Service, job
-from middlewared.utils import filter_list
 from middlewared.plugins.pwenc import PWENC_FILE_SECRET
 from middlewared.plugins.cluster_linux.utils import CTDBConfig, FuseConfig
 from middlewared.plugins.filesystem_ import stat_x
+from middlewared.schema import accepts, Bool, Dict, Float, Int, List, Ref, returns, Path, Str
+from middlewared.service import private, CallError, filterable_returns, Service, job
+from middlewared.utils import filter_list, run
 
 
 class FilesystemService(Service):
 
     class Config:
         cli_namespace = 'storage.filesystem'
 
+    @private
+    async def add_immutable_flag_to_path(self, path):
+        cp = await run(['chattr', '+i', '-RV', path], check=False)
+        if cp.returncode:
+            raise CallError(f'Unable to set immutable flag on at {path!r} path: {cp.stderr.decode()}')
+
+    @private
+    async def remove_immutable_flag_to_path(self, path):
+        cp = await run(['chattr', '-i', '-RV', path], check=False)
```

Comment:
Why subprocess when os module has chflags?

Reply:
Because we would like to do this recursively :)

Reply:
We don't need to do this recursively, it's an empty directory. If it is not empty for some reason (already contains user data due to some bug), we don't want to touch that.

Reply:
@themylogin so should we just make sure that we are not able to add/delete any files/folder in the mountpoint path ? Or should we also be making sure that any existing files there are not modified. Also nested directories contents can still be added/removed if it's not done recursively.

Reply:
@sonicaj that would be messing with user data, they might already have immutable attributes applied to some subset of files, we'll wipe all that. Having data in the mountpoint already is an emergency situation that will be resolved once the dataset is unlocked back and ideally should never happen again after we merge this PR. Doing this recursively might have sense, but I doubt this is really necessary.